lib/common/xml.c

```diff
@@ -2645,8 +2645,8 @@ sorted_xml(xmlNode *input, xmlNode *parent, gboolean recursive)
     pcmk_nvpairs2xml_attrs(nvpairs, result);
     pcmk_free_nvpairs(nvpairs);
 
-    for (child = pcmk__xml_first_child(input); child != NULL;
-         child = pcmk__xml_next(child)) {
+    for (child = pcmk__xe_first_child(input); child != NULL;
+         child = pcmk__xe_next(child)) {
```

Comment:
This looks like it preserves the existing behavior, but the existing behavior seems suboptimal -- comments and text nodes will be stripped when sorting XML attributes. However I believe we only use this on XML we create, that doesn't have any, so it should be fine to leave as is.

Reply:
I've been leaning towards caution on this PR, so if we're unsure, switching it back to pcmk__xml_first_child would be fine with me.

Reply:
Actually `input` has to be an element since we grab its name for `create_xml_node()`, which would make `result` NULL, so it's fine how you have it. (It might be nice to handle non-elements for future use, but not worth any effort now.)